src/EnergyPlus/FaultsManager.cc

```diff
@@ -1764,6 +1764,102 @@ namespace FaultsManager {
             FouledCoils(jFault_FoulingCoil).Rfa = rNumericArgs(3);
             FouledCoils(jFault_FoulingCoil).Aout = rNumericArgs(4);
             FouledCoils(jFault_FoulingCoil).Aratio = rNumericArgs(5);
+
+
+            // Coil check and link
+            {
+                // Obtains and Allocates WaterCoil related parameters from input file
+                if (WaterCoils::GetWaterCoilsInputFlag) {
+                    WaterCoils::GetWaterCoilInput();
+                    WaterCoils::GetWaterCoilsInputFlag = false;
+                }
+
+                // Check the coil name and type
+                int CoilNum = UtilityRoutines::FindItemInList(FouledCoils(jFault_FoulingCoil).FouledCoilName, WaterCoils::WaterCoil);
```

Comment:
I like that you are turning this around and more properly linking the fault with the coil.  If we proceed with adding more faults later, it would be good to give this a whole new design where the faults are classes with methods, and we can move away from this passive index linking.  But anyway, you've done a lot of good with this.  The faults should, well, work now!